### PR TITLE
Fix trace span logic for Python base SDK

### PR DESCRIPTION
**Pull request by selcukcihan**
### Description
* Related issue https://linear.app/serverless/issue/SC-356/aws-lambda-sdk-create-lambda-process-trace-spans-with-related-tags
* Fix trace timer by using the high resolution nanosecond timer
* Fix trace span close logic such that the context is updated to the first open ancestor, in case the closing span is the current span.
* Add base sdk to vscode python path so that intellisense works.

### Testing done
I've tested it on the cloud and was able to create traces together with the aws-serverless-sdk (which is wip, PR dedicated to it will be provided separately).

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,4 @@
 {
-    "cSpell.words": [
```
Comment (medikoo) [inline .vscode/settings.json]: Hmm.. actually, this file should not be a part of the repository, it landed here by mistake (when migrating documentation from other repo)

VS Code settings is mostly about personal settings (and it also implies VS Code as an editor choice, which also should be up to developer)

I've removed it from the repo. You can ensure it doesn't end in your git diff by updating gitignore global rules (see https://gist.github.com/subfuzion/db7f57fff2fb6998a16c)

**Review by medikoo — CHANGES_REQUESTED** [review]
Great find @selcukcihan all is good, let's just remove this .vscode update

File: .vscode/settings.json (hunk 1 above)
Comment (selcukcihan) [reply .vscode/settings.json]: Yeah makes sense 👍

**Review by medikoo — DISMISSED** [review]
Looks great 👍

**Review by medikoo — COMMENTED** [review]
@selcukcihan Again I have "This branch cannot be rebased due to conflicts", can you rebase the branch?

**Comment by selcukcihan** [comment]
> @selcukcihan Again I have "This branch cannot be rebased due to conflicts", can you rebase the branch?

Can you try again please? I've squashed the commits.

**Review by medikoo — APPROVED** [review]

**Pull request merged**